internal/localdebug/local_ide_debug.go

```diff
@@ -0,0 +1,263 @@
+package localdebug
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+const jvmDebugConfig = `<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="%s" type="Remote">
+    <option name="USE_SOCKET_TRANSPORT" value="true" />
+    <option name="SERVER_MODE" value="false" />
+    <option name="SHMEM_ADDRESS" />
+    <option name="HOST" value="localhost" />
+    <option name="PORT" value="%d" />
+    <option name="AUTO_RESTART" value="false" />
+    <RunnerSettings RunnerId="Debug">
+      <option name="DEBUG_PORT" value="%d" />
+      <option name="LOCAL" value="false" />
+    </RunnerSettings>
+    <method v="2" />
+  </configuration>
+</component>`
+
+const golangDebugConfig = `<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="%s" type="GoRemoteDebugConfigurationType" factoryName="Go Remote" port="%d">
+    <option name="disconnectOption" value="LEAVE" />
+    <disconnect value="LEAVE" />
+    <method v="2" />
+  </configuration>
+</component>`
+
+func init() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGINT)
+	signal.Notify(c, syscall.SIGTERM)
+	go func() {
+		<-c
+		lock.Lock()
+		defer lock.Unlock()
+		for p := range paths {
+			println(p)
+			_ = os.Remove(p)
+		}
+
+	}()
+}
+
+var lock = sync.Mutex{}
```

Comment:
These should definitely not be globals, it makes it difficult to test and maintain. Wrap all this, including the locking and signal handling, in a struct instead.

Reply:
The signal handling is fine yeah. Multiple listeners can be created.

Although, now that you mention it it would be preferable to have a `Close() error` method, and call that from the existing top-level signal handler in `main`.

Reply:
Thinking about it I can just use the context being cancelled.

Reply:
Even better!